index.js

```diff
@@ -49,38 +49,46 @@ module.exports = class PrettierPlugin {
 
   apply(compiler) {
     compiler.hooks.emit.tapAsync('Prettier', (compilation, callback) => {
-
-    const promises = [];
-    compilation.fileDependencies.forEach(filepath => {
-      if (this.extensions.indexOf(path.extname(filepath)) === -1) {
-        return;
-      }
-
-      if (/node_modules/.exec(filepath)) {
-        return;
-      }
-      promises.push(new Promise((resolve, reject) => {
-        fs.readFile(filepath, this.encoding, (err, source) => {
-          if (err) {
-              return reject(err);
-          }
-          const prettierSource = prettier.format(source, Object.assign({}, this.prettierOptions, { filepath }));
-          if (prettierSource !== source) {
-            fs.writeFile(filepath, prettierSource, this.encoding, err => {
-              if (err) {
+      const promises = [];
+      compilation.fileDependencies.forEach(filepath => {
+        if (this.extensions.indexOf(path.extname(filepath)) === -1) {
+          return;
+        }
+  
+        if (/node_modules/.exec(filepath)) {
+          return;
+        }
+        promises.push(new Promise((resolve, reject) => {
+          fs.readFile(filepath, this.encoding, (err, source) => {
+            if (err) {
                 return reject(err);
+            }
+            
+            try{
+              const prettierSource = prettier.format(source, Object.assign({}, this.prettierOptions, { filepath }));
+              if (prettierSource !== source) {
+                fs.writeFile(filepath, prettierSource, this.encoding, err => {
+                  if (err) {
+                    return reject(err);
+                  }
+                  resolve();
+                });
+              } else {
+                resolve();
               }
-              resolve();
-            });
-          } else {
-            resolve();
-          }
-        });
-      }));
-    });
-
-    Promise.all(promises).then(() => {
+            }
+            catch(err){
+              return reject(err);
+            }
+          });
+        }));
+      });
+  
+      Promise.all(promises).then(() => {
         callback();
+      }).catch(err => {
+        //compilation.errors.push(err);
```

Comment:
Did you mean to leave this comment in, or was this a debug thing?

Reply:
I was reading [here](https://stackoverflow.com/questions/42364742/webpack-plugin-error-management) that webpack a way of adding compilation errors as well (not just fatal errors). 

But I decided to not use that approach and pass the error object to the callback because all the Promise rejections are coming from fatal errors in your code, that way webpack will take it as a fatal and not compilation error.

I forgot to delete the comment. :flushed:

Reply:
I updated the PR, thanks!